src/t_cose_util.c

```diff
@@ -363,6 +384,178 @@ create_enc_structure(const char            *context_string,
 }
 
 
+/* The context information structure is used to ensure that the
+ * derived keying material is "bound" to the context of the transaction.
+ *
+ * The structure described below is based on Section 5.2 of RFC 9053
+ * with further details added in draft-ietf-suit-firmware-encryption.
+ *
+ * The following CDDL describes the content of the context information
+ * structure used for ES-DH.
+ *
+ * identity-kid     = 1
+ * identity-x5u     = 2
+ * identity-x5t     = 3
+ * identity-x5chain = 4
+ * identity-x5bag   = 5
+ *
+ * identity-type-id /= identity-kid
+ * identity-type-id /= identity-x5u
+ * identity-type-id /= identity-x5t
+ * identity-type-id /= identity-x5chain
+ * identity-type-id /= identity-x5bag
+ *
+ * Identity = [
+ *   type : identity-type-id,
+ *   // content of the parameter in the COSE structure of the indicated <type>
+ *   bytes : bstr,
+ * ]
+ *
+ * PartyInfoSender = (
+ *     identity : bstr "author"
+ *     nonce : nil,
+ *     other : bstr .cbor Identity
+ * )
+ *
+ * PartyInfoRecipient = (
+ *     identity : bstr "recipient",
+ *     nonce : nil,
+ *     other : bstr .cbor Identity
+ * )
+ *
+ * COSE_KDF_Context = [
+ *     // algorithm ID for payload encryption (e.g. AES-GCM-256)
+ *     AlgorithmID : int,
+ *     PartyUInfo : [ PartyInfoSender ],
+ *     PartyVInfo : [ PartyInfoRecipient ],
+ *     SuppPubInfo : [
+ *         // Length of the algorithm indicated in the AlgorithmID
+ *         keyDataLength : uint,
+ *         // serialized content of the recipients-inner.protected field
+ *         // e.g. h'a1013818' from Figure 7 in draft-ietf-suit-firmware-encryption-11
+ *         protected : bstr .cbor recipients-inner.protected,
+ *         other: bstr .size 0
+ *     ],
+ *     // hash of the encrypted payload/firmware in form of a SUIT_Digest structure
+ *     SuppPrivInfo : bstr .cbor SUIT_Digest
+ * ]
+ *
+ */
+enum t_cose_err_t
+create_info_structure(int32_t enc_alg,
```

Comment:
This seems like a good start, but it probably needs to be more general in relation so anything allowed in 9053 can be implemented. For example, maybe not hard code "author". More thinking needed.